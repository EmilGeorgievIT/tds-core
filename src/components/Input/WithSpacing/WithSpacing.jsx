import React from 'react'
import PropTypes from 'prop-types'

import styles from './WithSpacing.modules.scss'

const WithSpacing = ({ amount, children }) => (
  <div className={styles[`evenMarginBottom-${amount}`]}>
    {children}
  </div>
)
WithSpacing.propTypes = {
  amount: PropTypes.oneOf([1]).isRequired,
  children: PropTypes.node.isRequired
}

export default WithSpacing
