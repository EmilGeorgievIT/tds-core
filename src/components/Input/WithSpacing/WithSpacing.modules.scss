@import '../../../scss/settings/spacing';

.flexColumn {
  display: flex;
  flex-direction: column;
}

.evenMarginBottom-1 {
  composes: flexColumn;

  > * {
    margin-bottom: $spacing-x-tight;

    &:last-child {
      margin-bottom: 0;
    }
  }
}
