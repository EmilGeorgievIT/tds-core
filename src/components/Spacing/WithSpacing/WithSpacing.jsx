import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

import styles from './WithSpacing.modules.scss'

const capitalize = string => `${string.charAt(0).toUpperCase()}${string.slice(1)}`

const WithSpacing = ({ location, amount, children, ...rest }) => (
  <div {...safeRest(rest)} className={styles[`margin${capitalize(location)}-${amount}`]}>
    {children}
  </div>
)
WithSpacing.propTypes = {
  location: PropTypes.oneOf(['left', 'bottom']).isRequired,
  amount: PropTypes.oneOf([2, 3]).isRequired,
  children: PropTypes.node.isRequired
}

export default WithSpacing
