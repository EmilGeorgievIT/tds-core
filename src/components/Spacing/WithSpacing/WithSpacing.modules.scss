@import '../../../scss/settings/spacing';

.marginBottom-2 {
  margin-bottom: $spacing-tight;
}

.marginBottom-3 {
  margin-bottom: $spacing-base;
}
