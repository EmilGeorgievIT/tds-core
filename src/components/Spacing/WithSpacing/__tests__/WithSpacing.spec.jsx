import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import WithSpacing from '../WithSpacing'

describe('WithSpacing', () => {
  const defaultProps = {
    amount: 2
  }
  const doShallow = (props = {}) => (
    shallow(<WithSpacing {...defaultProps} {...props}>Some content</WithSpacing>)
  )

  it('renders', () => {
    const flexbox = doShallow()

    expect(toJson(flexbox)).toMatchSnapshot()
  })

  it('applies margin bottom', () => {
    const flexbox = doShallow({ amount: 3 })

    expect(flexbox).toHaveClassName('marginBottom-3')
  })

  it('passes additional attributes to the HTML element', () => {
    const flexbox = doShallow({ id: 'the-id', 'data-some-property': 'a value' })

    expect(flexbox).toHaveProp('id', 'the-id')
    expect(flexbox).toHaveProp('data-some-property', 'a value')
  })

  it('does not allow custom CSS', () => {
    const flexbox = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(flexbox).not.toHaveProp('className', 'my-custom-class')
    expect(flexbox).not.toHaveProp('style')
  })
})
