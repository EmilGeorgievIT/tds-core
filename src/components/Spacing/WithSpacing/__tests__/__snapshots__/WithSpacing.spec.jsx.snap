// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSpacing renders 1`] = `
<div
  className="marginBottom-2"
>
  Some content
</div>
`;
